src/java/com/phenix/pct/PCTNSBroker.java

```diff
@@ -0,0 +1,252 @@
+     * 
+     * @throws BuildException Something went wrong during write.
+     * 
+     */
+    private void writeDeltaFile() {
+        try (Writer w = new FileWriter(tmpFile); PrintWriter bw = new PrintWriter(w)) {
+            bw.println("[NameServer." + this.nameServer + "]");
+            if (!"delete".equalsIgnoreCase(this.action)) {
+                bw.println("environment=" + this.name);
+                bw.println("autoStart=" + (this.autoStart ? "1" : "0"));
+                bw.println("brokerKeepAliveTimeout=" + this.brokerKeepAliveTimeout);
+                bw.println("logAppend=" + (this.logAppend ? "1" : "0"));
+                if (this.portNumber != -1)
+                    bw.println("portNumber=" + this.portNumber);
+                if (this.serverLogFile != null)
+                    bw.println("srvrLogFile=" + this.serverLogFile);
+                if (this.serverLogLevel != -1)
+                    bw.println("srvrLoggingLevel=" + this.serverLogLevel);
+                if (this.workDir != null)
+                    bw.println("workDir=" + this.workDir);
+
+            }
+            bw.close();
+        } catch (IOException caught) {
+            throw new BuildException("Unable to create temp file", caught);
+        }
+    }
+
+    /**
+     * Creates an Exec task, calling mergeprop shell script.
+     */
+    private Task getCmdLineMergeTask(File propFile, File deltaFile) {
+        ExecTask task = new ExecTask(this);
+        task.setDir(this.getProject().getBaseDir());
+        task.setExecutable(this.getExecPath("mergeprop").getAbsolutePath());
+        task.setFailonerror(true);
+
+        Environment.Variable var4 = new Environment.Variable();
+        var4.setKey("DLC"); //$NON-NLS-1$
+        var4.setValue(this.getDlcHome().toString());
+        task.addEnv(var4);
+
+        task.createArg().setValue("-type");
+        task.createArg().setValue("ubroker");
+        task.createArg().setValue("-action");
+        task.createArg().setValue(this.action);
+        task.createArg().setValue("-target");
+        task.createArg().setValue(propFile.getAbsolutePath());
+        task.createArg().setValue("-delta");
+        task.createArg().setValue(deltaFile.getAbsolutePath());
+
+        return task;
+    }
+
+    /**
+     * Cross-attributes check
+     * 
+     * @throws BuildException Attributes are wrong...
+     */
+    private void checkAttributes() throws BuildException {
+        if ((!action.equalsIgnoreCase(UPDATE)) && (!action.equalsIgnoreCase(CREATE))
+                && (!action.equalsIgnoreCase(DELETE))) {
+            throw new BuildException("Unknown action : " + action);
+        }
+
+        if ((serverLogLevel != -1) && ((serverLogLevel < 1) || (serverLogLevel > 5)))
+            throw new BuildException("Log level should be between 1 and 5");
+
+        if (this.name == null) {
+            throw new BuildException("Name attribute is missing");
+        }
+        if (this.action == null) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "false" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2583)